cfme/tests/automate/test_domain.py

```diff
@@ -182,8 +182,15 @@ def test_domain_name_wrong(appliance):
         initialEstimate: 1/60h
         tags: automate
     """
-    with pytest.raises(Exception, match='Name may contain only'):
-        appliance.collections.domains.create(name='with space')
+    wrong_domain = 'Dummy Domain'
+    domain = appliance.collections.domains
+    view = navigate_to(domain, 'Add')
```

Comment:
Optional:
@ganeshhubale I think better to go with create domain and use `pytest.raise(AssiersionError)`. As `flash.assert_no_error()` should raise this error. This comment totally optional.

Reply:
I agree with this ^ Ganesh . 
Please create domain using collection 

     domain = appliance.collections.domains.create(name=wrong_domain, description=fauxfactory.gen_alpha(),
                                                                                            enabled=enabled)
and then check for flash mesage
and 
assert not  domain.exists

Reply:
@digitronik @sshveta 
- I am agree with both of you.
- Please see this discussion, as per [this comment](https://github.com/ManageIQ/integration_tests/pull/8443#discussion_r259083403) by @jawatts ; I too think that it's good to go with GUI testing while creating domain/namespace.
- Also sometimes what happens is that, while creating ```class``` with invalid name it gives flash message like this: **Error during 'add': Validation failed: MiqAeClass: Name may contain only alphanumeric and _ . - characters**
- As I see this is error, It might be bug. So I have added comment for it in [BZ](https://bugzilla.redhat.com/show_bug.cgi?id=1650071#c14)
- Previously it was as you both said.
- I would be happy to update with suggestions from all of you @sshveta @digitronik @jawatts 

Please let me know, which one is best way for this test case.

Thank you,
Ganesh

Reply:
@ganeshhubale @digitronik @sshveta So I agree with @digitronik @sshveta if we were not checking for a specific error message. It seems like we have a BZ around the error message, so I would argue we should explicitly be checking the flash message in this case and not just an `AssertionError`

Reply:
My view on this:
@ganeshhubale @jawatts @sshveta really its debatable matter. As per me it should contributor privilege so I added `Optional` comment. I don't want to manipulate his/her way of codding as both way are correct as per me.

@sshveta your comment is also valid:
1. Its negative test case.... we are expecting some error mainly `AssertionError` due to `assert_no_error()`. I remember my patch with `Mike` guidance that this method should allow only error not `warning` or other.
2. As `pytest.raise` capture error it will come out of method and ready for other operation. So @sshveta suggestion means asserting exact flash will work as `flash` is part of `BaseEntityView` we can catch from any view.

Reply:
@digitronik agree with you here. 

@ganeshhubale I think the takeaway here for you, code it how you feel is correct (both ways here are correct). But we should check the explicit flash message and make sure the object is not created.

Reply:
Thanks for suggestions @digitronik @sshveta @digitronik 
I will check ```AssertionError``` using ```pytest.raise``` then will check ```flash``` message and ```domain.exist``` or not.